packages/expo/scripts/resolveAppEntry.js

```diff
@@ -29,20 +28,17 @@ if (!platform || !projectRoot) {
   process.exit(1);
 }
 
-const entry = resolveEntryPoint(projectRoot, { platform });
+const entry = absolute
+  ? resolveEntryPoint(projectRoot, { platform })
+  : resolveRelativeEntryPoint(projectRoot, { platform });
 
-if (entry) {
+if (!entry) {
+  console.error(`Error: Could not find entry file for project at: ${projectRoot}`);
+  process.exit(1);
+} else {
   // Prevent any logs from the app.config.js
   // from being used in the output of this command.
   console.clear();
-  // React Native's `PROJECT_ROOT` could be using a different root on MacOS (`/var` vs `/private/var`)
-  // We need to make sure to get the real path, `resolveEntryPoint` is using this too
-  console.log(
-    absolute
-      ? path.resolve(entry)
-      : path.relative(fs.realpathSync(projectRoot), fs.realpathSync(entry))
-  );
-} else {
-  console.error(`Error: Could not find entry file for project at: ${projectRoot}`);
-  process.exit(1);
+
+  console.log(absolute ? path.resolve(entry) : entry);
```

Comment:
`resolveEntryPoint` always returns an absolute path, I don't think `path.resolve` is necessary here. But we can even refactor this a bit more with [the new `convertEntryPointToRelative` method](https://github.com/expo/expo/pull/30654/files#diff-93c9336391d70f2bfce1c14e44e5fb0e8339a66e8b56212da5dc7bc902eb3b8aR134).

```suggestion
  console.log(entry);
```